lnx-engine/search-index/src/schema.rs

```diff
@@ -0,0 +1,538 @@
+use std::iter::FromIterator;
+
+use anyhow::{anyhow, Error, Result};
+use hashbrown::{HashMap, HashSet};
+use serde::{Deserialize, Serialize};
+use tantivy::schema::{
+    Cardinality,
+    FacetOptions,
+    Field,
+    FieldType,
+    IndexRecordOption,
+    IntOptions,
+    Schema,
+    SchemaBuilder,
+    TextFieldIndexing,
+    TextOptions,
+    FAST,
+    INDEXED,
+    STORED,
+};
+use tantivy::Score;
+
+use crate::helpers::{Calculated, Validate};
+
+pub static PRIMARY_KEY: &str = "_id";
+
+fn default_to_true() -> bool {
+    true
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SchemaContext {
+    /// The index's fields.
+    ///
+    /// Document entries will be implicitly converted into the index's schema types
+    /// if they need to be.
+    fields: HashMap<String, FieldDeclaration>,
+
+    /// The fields what are actually searched via tantivy.
+    ///
+    /// These values need to either be a fast field (ints) or TEXT.
+    search_fields: Vec<String>,
+
+    /// A set of fields to boost by a given factor.
+    #[serde(default)]
+    boost_fields: HashMap<String, Score>,
+
+    #[serde(skip)]
+    required_fields: HashSet<String>,
+
+    #[serde(skip)]
+    multi_value_fields: HashSet<String>,
+}
+
+impl Validate for SchemaContext {
+    fn validate(&self) -> Result<()> {
+        if self.search_fields.is_empty() {
+            return Err(Error::msg(
+                "at least one indexed field must be given to search.",
```

Comment:
Does it make sense to make search_fields optional and by default use all indexed fields for search?

Reply:
Thats a good idea! Shall add that.